aea/cli/config.py

```diff
@@ -54,89 +63,269 @@ def get(ctx: Context, json_path: List[str]):
     type=click.Choice(["str", "int", "bool", "float"]),
     help="Specify the type of the value.",
 )
-@click.argument("JSON_PATH", required=True, type=AEAJsonPathType())
+@click.argument("JSON_PATH", required=True)
 @click.argument("VALUE", required=True, type=str)
 @pass_ctx
 def set_command(
     ctx: Context,
-    json_path: List[str],
+    json_path: str,
     value: str,
     type: str,  # pylint: disable=redefined-builtin
 ):
     """Set a field."""
-    _set_config(ctx, json_path, value, type)
+    ConfigGetSet(ctx, json_path).set(value, type)
+
 
+class ConfigGetSet:
+    """Tool to get/set value in agent config."""
 
-def _get_config_value(ctx: Context, json_path: List[str]):
-    config_loader = cast(ConfigLoader, ctx.config.get("configuration_loader"))
-    configuration_file_path = cast(str, ctx.config.get("configuration_file_path"))
+    FIELDS_NOT_ALLOWED_TO_CHANGE = [
+        "name",
+        "version",
+        "author",
+        "type",
+        "connections",
+        "skills",
+        "protocols",
+        "contracts",
+    ]
 
-    configuration_object = load_yaml(configuration_file_path)
-    config_loader.validator.validate(instance=configuration_object)
+    AGENT_FIELDS_ALLOWED_TO_CHANGE = [
+        "agent_name",
+        "author",
+        "registry_path",
+        "logging_config",
+        "private_key_paths",
+        "default_connection",
+        "default_ledger",
+    ]
 
-    parent_object_path = json_path[:-1]
-    attribute_name = json_path[-1]
-    parent_object = _get_and_validate_parent_obj(
-        configuration_object, parent_object_path, attribute_name
-    )
+    def __init__(self, ctx: Context, dotted_path: str) -> None:
+        """Init tool.
 
-    return parent_object.get(attribute_name)
+        :param ctx: click context
+        :param dotted_path: str with attribute path to get/set
+        """
+        self.dotted_path = dotted_path
+        self.ctx = ctx
 
+        (
+            self.json_path,
+            self.path_to_resource_configuration,
+            self.config_loader,
+            self.component_id,
+        ) = self._handle_dotted_path()
+
+    def _handle_dotted_path(
+        self,
+    ) -> Tuple[List[str], Path, ConfigLoader, Optional[ComponentId]]:
+        """Handle dotted path."""
+        try:
+            return handle_dotted_path(self.dotted_path, self.agent_config.author)
+        except AEAException as e:
+            raise click.ClickException(*e.args)
+
+    @property
+    def attr_name(self) -> str:
+        """Attribute name."""
+        return self.json_path[-1]
+
+    def get(self) -> Any:
```

Comment:
rather than Any, can you define the union of valid types please?

Reply:
it uses dict get operation, there cane be any value, depends on how the component config was loaded. it's ok to assume you dont know the type is going to be returned